src/ucp/core/ucp_ep.c

```diff
@@ -2749,10 +2752,22 @@ void ucp_ep_reqs_purge(ucp_ep_h ucp_ep, ucs_status_t status)
          * context) and not invalidated yet, also remote EP ID is already set */
         !(ucp_ep->flags &
           (UCP_EP_FLAG_ON_MATCH_CTX | UCP_EP_FLAG_CLOSE_REQ_VALID))) {
-        ucs_hlist_for_each_extract(req, &ucp_ep_flush_state(ucp_ep)->reqs,
-                                   send.list) {
-            ucp_ep_flush_request_ff(req, status);
+        flush_state = ucp_ep_flush_state(ucp_ep);
+        ucs_hlist_for_each_extract(req, &flush_state->reqs, send.list) {
+            ucp_ep_flush_request_ff(req, status, 0);
         }
+
+        /* Adjust 'comp_sn' value to a value stored in 'send_sn' because those
+         * uncompleted operations won't be completed anymore */
+        ucs_assert(UCS_CIRCULAR_COMPARE32(flush_state->cmpl_sn, <=,
+                                          flush_state->send_sn));
+        num_comps            = flush_state->send_sn - flush_state->cmpl_sn;
+        flush_state->cmpl_sn = flush_state->send_sn;
+
+        /* Adjsut 'flush_ops_count' value to a number of uncompleted operations
+         * which are waiting for remote completions in this flush state */
+        ucs_assert(worker->flush_ops_count >= num_comps);
+        worker->flush_ops_count -= num_comps;
```

Comment:
this code actually emulates a CPML packet (which did not really arrive)
so let's reuse code with whatever is being done when processing CMPL packet

Reply:
@yosefe do you mean to use `ucp_ep_rma_remote_request_completed()`?
but it adjusts `flush_state->cmpl_sn` and then go over all requests and check whether they need to be completed or not.
I'm not sure that it will be optimal to use the same logic here.
IMO beter to just reset all counters

Reply:
yes, i think we can reuse ucp_ep_rma_remote_request_completed, or at least extract common code with it

Reply:
@yosefe unfortunately, we can't get rid of the code which adjusts cmpl_sn and flush_count counters after the loop over all requests under flush. It is impossible since PUT SW implementation don't wait for completion and just completes local side after sending the data to a peer. So, RMA SW completions are received after UCP requests are completed. This completion packet is needed to update flush only.

Reply:
in any case, we should use ucp_ep_flush_remote_completed() to emulate remote CMPL, and not ucp_ep_flush_request_ff() [which is for pending purge flow only]